vault/mount.go

```diff
@@ -683,10 +683,17 @@ func (c *Core) remount(ctx context.Context, src, dst string) error {
 	}
 
 	// Verify exact match of the route
-	match := c.router.MatchingMount(ctx, src)
-	if match == "" || ns.Path+src != match {
+	srcMatch := c.router.MatchingMountEntry(ctx, src)
+	if srcMatch == nil {
 		return fmt.Errorf("no matching mount at %q", src)
 	}
+	if srcMatch.NamespaceID != ns.ID {
+		return fmt.Errorf("source mount in a different namespace than request")
+	}
+
+	if err := verifyNamespace(c, ns, &MountEntry{Path: dst}); err != nil {
+		return err
+	}
```

Comment:
Do we need to do the same below to make sure `dst` is in the correct namespace?

Reply:
Do what same? Aren't we already checking that?

Reply:
```
	if dstMatch.NamespaceID != ns.ID {
		return fmt.Errorf("destination mount in a different namespace than request")
	}
```

Reply:
`verifyNamespace` handles it

Reply:
Also note that `dstMatch` will be nil or we won't proceed anyways so there's no `dstMatch.NamespaceID` to look at. Hence using `verifyNamespace`.

Reply:
Eh, i guess that's not correct since any existence of a mount is an error. But are we verifying that `dst` isn't pointing to a child namespace?

Reply:
Yes, `verifyNamespace` does that.